grind/python/disttest/isolate.py

```diff
@@ -97,14 +98,18 @@ def _generate_run_script_contents(self):
         exit $rc
     fi
 }
-export JAVA7_BUILD=true
-. /opt/toolchain/toolchain.sh
+
 run ./unpack.sh # Generated by packager
 source environment.source # Init runtime environment
+"""
+        contents += """
+jdks=(/usr/lib/jvm/java-{java_version}-*)
+export JAVA_HOME=${{jdks[0]}}
+""".format(java_version=self.java_version)
+
+        contents += """
 run which mvn
 run mvn -version
-run which java
```

Comment:
The java executable in the PATH is not used by Maven which relies on JAVA_HOME. BTW, "mvn -version" prints Java version and jdk path:

> 2016-10-21 23:02:33 Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T11:57:37+00:00)
> 2016-10-21 23:02:33 Maven home: /tmp/run_tha_testYcMYUa/apache-maven-3.3.3
> 2016-10-21 23:02:33 Java version: 1.8.0_91, vendor: Oracle Corporation
> 2016-10-21 23:02:33 Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
> 2016-10-21 23:02:33 Default locale: en_US, platform encoding: ANSI_X3.4-1968
> 2016-10-21 23:02:33 OS name: "linux", version: "3.16.0-31-generic", arch: "amd64", family: "unix"